### DISCUSSION TITLE
Maximum text width accounting for sidebar toggle

### DISCUSSION THREAD

**Discussion by chchen2021** [Zettlr/Zettlr#2516] [Q&A] [answered]
The following [custom css](https://docs.zettlr.com/en/core/custom-css/) is very helpful in setting the maxim width of the body text, a width that is comfortable for writing and adapts to the screen size:
```
#editor {
  --side-margin: calc( 50vw - 30em ); 
}

#editor .CodeMirror {
  margin-left: var(--side-margin);
}

#editor .CodeMirror-scroll {
  padding-right: var(--side-margin);
}
```

How do I revise this css snippet to account for the extra width of the sidebar (and file manager)? I have a persistent file manager and a sidebar that I show occasionally. Whenever I click the sidebar, the text width becomes very narrow because the margins are fixed. The changing text width makes it difficult to edit and find things. 

The ideal behavior would be a fixed text width whether or not the sidebar is hidden. In Typora, I could set a persistent `max-width` under `#write`. I tried to impose max-width in Zettlr but no success.

**Comment by chchen2021** [comment]
I tried setting two different margins depending on if the sidebar is active, according to this [website](https://forum.freecodecamp.org/t/question-about-bootstrap-sidebar/208562), but it does not seem to work. What am I missing?

```
    #sidebar {
        margin-left: -250px;
    }

    #sidebar .active {
        margin-left: 0;
    }
```

By the way, I can set the maximum width within `#editor .CodeMirror {}`, but the fixed-width editor does not occupy the full screen. I am therefore looking at setting two different margins depending on the sidebar status.

**Reply by nathanlesage** [reply]
Zettlr doens't use bootstrap, so that's one reason. The other is that both the editor and the sidebar are in their own containers so it's going to be tricky anyhow ... what you MIGHT be able to achieve is do some calc magic, but even then I'm a little bit lost for ideas.

**Reply by chchen2021** [reply]
Thanks for trying to help on every issue I raised. I know this takes a lot of your time and really appreciate your help. 

I just noticed that the File Manager and Sidebar panes are of the same width if they appear one at a time, so I just need to toggle one of these panes to address the most pressing issue: "The changing text width makes it difficult to edit and find things." With a keyboard shortcut, the toggling can happen very quickly. This is a good enough solution for me.

**Comment by chchen2021** [comment, marked as answer]
A good solution is to set the same text width, one for distraction-free mode, the other for the normal mode. The following CSS works on a 2880 x 1800 MacBook Pro (connected to a 2560 x 1440 monitor). See also [#244](https://github.com/Zettlr/Zettlr/issues/244) on the CSS code for the distraction-free mode.
```
:root {
  --side-margin: calc(31.5vw - 18em); 
  --side-margin-fullscreen: calc(41.5vw - 18em);
}

@media only screen and (min-width: 1000px) {
  #editor .CodeMirror {
    margin-left: var(--side-margin);
  }
  
  #editor .CodeMirror-scroll {
    padding-right: var(--side-margin);
  }    
    
  #editor.fullscreen .CodeMirror-fullscreen {
    margin-left: var(--side-margin-fullscreen) !important;
  }
    
  #editor.fullscreen .CodeMirror-fullscreen .CodeMirror-scroll { 
    padding-right: var(--side-margin-fullscreen) !important; 
   }
}
```
